Index: gui/pakselector.cc
===================================================================
--- gui/pakselector.cc	(Revision 3130)
+++ gui/pakselector.cc	(Arbeitskopie)
@@ -73,7 +73,7 @@
 bool pakselector_t::check_file( const char *filename, const char * )
 {
 	char buf[1024];
-	sprintf( buf, "%s/ground.Outside.pak", filename );
+	sprintf( buf, "%s/ground.Outside_.pak", filename );
 	FILE *f=fopen( buf, "r" );
 	if(f) {
 		fclose(f);
Index: simconst.h
===================================================================
--- simconst.h	(Revision 3130)
+++ simconst.h	(Arbeitskopie)
@@ -21,7 +21,7 @@
 //#define DOUBLE_GROUNDS
 
 /* single height is only 8 pixel (default 16) */
-//#define HALF_HEIGHT
+#define HALF_HEIGHT
 
 /* construct automatic bridges also as active player */
 //#define AUTOMATIC_BRIDGES
